packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/Abs_L2DepositedToken.sol

```diff
@@ -139,66 +140,98 @@ abstract contract Abs_L2DepositedToken is iOVM_L2DepositedToken, OVM_CrossDomain
 
     /**
      * @dev initiate a withdraw of some tokens to the caller's account on L1
-     * @param _amount Amount of the token to withdraw
+     * @param _amount Amount of the token to withdraw.
+     * @param _data Data to forward to L1.
+     * @param _l1Gas Gas limit for the provided message.
      */
     function withdraw(
-        uint _amount
+        uint _amount,
+        bytes calldata _data,
+        uint32 _l1Gas
     )
         external
         override
         virtual
         onlyInitialized()
     {
-        _initiateWithdrawal(msg.sender, _amount);
+        _initiateWithdrawal(
+            msg.sender,
+            msg.sender,
+            _amount,
+            _data,
+            _l1Gas
+        );
     }
 
     /**
-     * @dev initiate a withdraw of some token to a recipient's account on L1
-     * @param _to L1 adress to credit the withdrawal to
-     * @param _amount Amount of the token to withdraw
+     * @dev initiate a withdraw of some token to a recipient's account on L1.
+     * @param _to L1 adress to credit the withdrawal to.
+     * @param _amount Amount of the token to withdraw.
+     * @param _data Data to forward to L1.
+     * @param _l1Gas Gas limit for the provided message.
      */
     function withdrawTo(
         address _to,
-        uint _amount
+        uint _amount,
+        bytes calldata _data,
+        uint32 _l1Gas
     )
         external
         override
         virtual
         onlyInitialized()
     {
-        _initiateWithdrawal(_to, _amount);
+        _initiateWithdrawal(
+            msg.sender,
+            _to,
+            _amount,
+            _data,
+            _l1Gas
+        );
     }
 
     /**
      * @dev Performs the logic for deposits by storing the token and informing the L2 token Gateway of the deposit.
      *
-     * @param _to Account to give the withdrawal to on L1
-     * @param _amount Amount of the token to withdraw
+     * @param _from Account to pull the deposit from on L2.
+     * @param _to Account to give the withdrawal to on L1.
+     * @param _amount Amount of the token to withdraw.
+     * @param _data Data to forward to L1.
+     * @param _l1Gas Gas limit for the provided message on L1.
      */
     function _initiateWithdrawal(
+        address _from,
         address _to,
-        uint _amount
+        uint _amount,
+        bytes calldata _data,
+        uint32 _l1Gas
```

Comment:
This l1 gas value is not used at all atm. Do we plan on reintroducing it?

Reply:
Hmmm, not even in the child contracts? I will investigate.

Reply:
I see it just below on line 226. Am I missing something?

Reply:
1. Yes, sorry :)
2. Oh... great call out, specifically here:

https://github.com/ethereum-optimism/optimism/blob/dcb538d459940a021c580ec33e96a663a972ee49/packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L2CrossDomainMessenger.sol#L149-L157

hmmm... 

The question now becomes, do we remove it from this contract, or add it to the OVM_L2ToL1MessagePasser. Since it is much easier to remove it from this one (and there is a `replayMessage()` function available in the L1 Messanger), I will remove it from this one in an atomic commit that can be easily reverted if the decision goes the other way.

Reply:
I am of the opinion that we should have the `l1Gas` be used in the future; this makes it more robust for relayer markets which should develop in the future.  I think that should be a separate/future PR though; I see this as extremely related to https://github.com/ethereum-optimism/optimism/issues/745, which is starting to shape up in my head as a "polish xDomain messaging" mini project.  For this reason I am a fan of doing what @maurelian lays out in 1.  I am also a fan of just removing the `get*Gas()` functions and having an argument.